<?xml version="1.0"?>
<launch>
    <arg name="robot" default="robot1"/>
    <arg name="R" default="255"/>
    <arg name="G" default="0"/>
    <arg name="B" default="0"/>
    <arg name="x" default="0"/>
    <arg name="y" default="0"/>
    <arg name="theta" default="0"/>
    <arg name="radius" default=".15"/>
    <arg name="zero_pose" default="false"/>
    
    <arg name="controller_frequency" default="5."/>
    <arg name="use_move_base" default="false"/>
    
    <arg name="R01" value="$(eval R/255.)"/>
    <arg name="G01" value="$(eval G/255.)"/>
    <arg name="B01" value="$(eval B/255.)"/>   
    
    
    <group ns="$(arg robot)">
        <param name="robot_description" command="$(find xacro)/xacro $(find mrs_monitor)/urdf/robot.xacro prefix:=$(arg robot)/ radius:=$(arg radius)  R:=$(arg R01) G:=$(arg G01) B:=$(arg B01)"/>
        
<!--         no need for a robot state publisher for this 1-link robot -->
<!--         <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher"/> -->
        

        <node pkg="map_simulator" type="spawn" name="spawn" output="screen">
            <param name="zero_joints" value="true"/>
            <param name="static_tf_odom" value="true"/>
            <param name="radius" value="$(arg radius)"/>
            <param name="x" value="$(arg x)" unless="$(arg zero_pose)"/>
            <param name="y" value="$(arg y)" unless="$(arg zero_pose)"/>
            <param name="theta" value="$(arg theta)" unless="$(arg zero_pose)"/>
            <param name="force_scanner" value="false"/>
            <param name="robot_color" value="$(arg R) $(arg G) $(arg B)"/>
        </node>
        
        <!-- Run lighter control if not use_move_base -->
        <node pkg="mrs_monitor" type="mrs_control" name="move_base" unless="$(arg use_move_base)" output="screen">
            <rosparam file="$(find mrs_monitor)/param/costmap_common.yaml" command="load" ns="local_costmap"  subst_value="true"/>
            <rosparam file="$(find mrs_monitor)/param/local_costmap.yaml" command="load"  subst_value="true"/>
            <rosparam file="$(find mrs_monitor)/param/dwa_local_planner.yaml" command="load" subst_value="true" />
        </node>
        
        <!-- Or run the full move_base stack -->
        <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" if="$(arg use_move_base)">
            <rosparam file="$(find mrs_monitor)/param/costmap_common.yaml" command="load" ns="global_costmap"  subst_value="true"/>
            <rosparam file="$(find mrs_monitor)/param/costmap_common.yaml" command="load" ns="local_costmap"  subst_value="true"/>
            <rosparam file="$(find mrs_monitor)/param/local_costmap.yaml" command="load"  subst_value="true"/>
            <rosparam file="$(find mrs_monitor)/param/global_costmap.yaml" command="load" subst_value="true"/>
            <rosparam file="$(find mrs_monitor)/param/move_base.yaml" command="load" />
            <rosparam file="$(find mrs_monitor)/param/dwa_local_planner.yaml" command="load" subst_value="true" />
            <rosparam file="$(find mrs_monitor)/param/planner.yaml" command="load" />
    </node>
        
    </group>
</launch>
